test/e2e/webhook_test.go

```diff
@@ -590,26 +592,74 @@ var _ = Describe("Fleet's CRD Resource Handler webhook tests", func() {
 	})
 })
 
-var _ = Describe("Fleet's CR Resource Handler webhook tests", func() {
-	Context("CR validation webhook", func() {
+var _ = Describe("Fleet's Custom Resource Handler webhook tests", func() {
+	BeforeEach(func() {
+		// Creating this MC for IMC E2E, this MC will fail to join since it's name is not configured to be recognized by the member agent
+		// which it uses to create the namespace to watch for IMC resource. But it serves its purpose for the tests.
+		mc := &fleetv1alpha1.MemberCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testMemberCluster,
+			},
+			Spec: fleetv1alpha1.MemberClusterSpec{
+				Identity: rbacv1.Subject{
+					Name:      "test-user",
+					Kind:      "ServiceAccount",
+					Namespace: utils.FleetSystemNamespace,
+				},
+				State:                  fleetv1alpha1.ClusterStateJoin,
+				HeartbeatPeriodSeconds: 60,
+			},
+		}
+		Eventually(func() error {
+			return HubCluster.KubeClient.Create(ctx, mc)
+		}, testutils.PollTimeout, testutils.PollInterval).Should(Succeed())
+
+		imc := &fleetv1alpha1.InternalMemberCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testMemberCluster,
+				Namespace: testMemberClusterNamespace,
+			},
+		}
+		Eventually(func() error {
+			return HubCluster.KubeClient.Get(ctx, types.NamespacedName{Name: imc.Name, Namespace: imc.Namespace}, imc)
+		}, testutils.PollTimeout, testutils.PollInterval).Should(Succeed())
+	})
+	AfterEach(func() {
+		mc := &fleetv1alpha1.MemberCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testMemberCluster,
+			},
+		}
+		Expect(HubCluster.KubeClient.Delete(ctx, mc))
+		imc := &fleetv1alpha1.InternalMemberCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testMemberCluster,
+				Namespace: testMemberClusterNamespace,
+			},
+		}
+		Eventually(func() bool {
+			return k8sErrors.IsNotFound(HubCluster.KubeClient.Get(ctx, types.NamespacedName{Name: imc.Name, Namespace: imc.Namespace}, imc))
+		}, testutils.PollTimeout, testutils.PollInterval).Should(BeTrue())
+	})
+	Context("fleet guard rail tests for MC, IMC", func() {
 		It("should deny CREATE operation on member cluster CR for user not in system:masters group", func() {
-			mc := fleetv1alpha1.MemberCluster{
+			mc := &fleetv1alpha1.MemberCluster{
 				ObjectMeta: metav1.ObjectMeta{
-					Name: "test-member-cluster",
+					Name: testMemberCluster,
```

Comment:
isn't this MC already created in the BeforeEach already? did I mis read something?